internal/worker/uniter/runner/context/context.go

```diff
@@ -1239,15 +1245,61 @@ func (ctx *HookContext) GetRawK8sSpec() (string, error) {
 // CloudSpec return the cloud specification for the running unit's model.
 // Implements jujuc.HookContext.ContextUnit, part of runner.Context.
 func (ctx *HookContext) CloudSpec() (*params.CloudSpec, error) {
-	if ctx.modelType == model.CAAS {
-		return nil, errors.NotSupportedf("credential-get on a %q model", model.CAAS)
+	if ctx.cloudSpec != nil {
```

Comment:
I think this is a desired change. It looked like `ctx.cloudSpec` was intended to be a cache of this value, but it was set and never used.